pkg/fleet-manager/application/controller.go

```diff
@@ -136,16 +136,19 @@ func (a *ApplicationManager) Reconcile(ctx context.Context, req ctrl.Request) (_
 
 	// there only one fleet, so pre-fetch it here.
 	fleetKey := generateFleetKey(app)
-	fleet := &fleetapi.Fleet{}
-	// Retrieve fleet object based on the defined fleet key
-	if err := a.Client.Get(ctx, fleetKey, fleet); err != nil {
-		if apierrors.IsNotFound(err) {
-			log.Info("fleet does not exist", "fleet", fleetKey)
-			return ctrl.Result{RequeueAfter: fleetmanager.RequeueAfter}, nil
+	var fleet *fleetapi.Fleet
+	if fleetKey.Name != "" {
```

Comment:
when this can be empty

Reply:
https://github.com/kurator-dev/kurator/blob/19220c18ea2ca59ca40fc68c98701da777fe4a99/pkg/fleet-manager/application/controller.go#L157
fleet is used here